scripts/pages/pgCategoryDetail.ts

```diff
@@ -69,90 +72,93 @@ export default class PgCategoryDetail extends PgCategoryDetailDesign {
-                        itemImage1: product1?.image || '',
-                        itemTag1: product1?.discountTag || '',
-                        itemReview1: !!product1?.review ? product1?.review : '',
-
-                        itemTitle2: product2?.name || '',
-                        itemDesc2: product2?.description || '',
-                        itemDiscountPrice2: !!product2?.discount ? `$${product2?.discount}` : '',
-                        itemPrice2: `$${product2?.price}` || '',
-                        itemImage2: product2?.image || '',
-                        itemTag2: product2?.discountTag || '',
-                        itemReview2: !!product2?.review ? product2?.review : ''
-                    })
-                );
-            }
-        }
-        return processorItems;
-    }
     getCategoryProducts() {
         this.categoryProducts = store.getState().products.filter((product) => product.categoryId === this.routeData.dataId);
     }
     getShowcaseProducts() {
         this.categoryProducts = store
             .getState()
             .showcaseProducts.find((showcase) => showcase.showcaseId === this.routeData.dataId).products;
+    }
 
-        console.log('this.categoryProducts', this.categoryProducts);
+    initGridView() {
+        this.gvProducts.onItemBind = (GridViewItem: GviProductItem, productIndex: number) => {
+            GridViewItem.itemTag = this.categoryProducts[productIndex].discountTag;
+            GridViewItem.itemTitle = this.categoryProducts[productIndex].name;
+            GridViewItem.itemDesc = this.categoryProducts[productIndex].description;
+            GridViewItem.itemImage = this.categoryProducts[productIndex].image;
+            GridViewItem.itemDiscountPrice = !!this.categoryProducts[productIndex].discount
+                ? `$${this.categoryProducts[productIndex].discount}`
+                : false;
+            GridViewItem.itemPrice = `$${this.categoryProducts[productIndex].price}`;
+            GridViewItem.itemReview = !!this.categoryProducts[productIndex].review ? this.categoryProducts[productIndex]?.review : false;
+            GridViewItem.onActionClick = () => {
+                GridViewItem.initIndicator();
+                GridViewItem.toggleIndicator(true);
+                store.dispatch({
+                    type: 'ADD_TO_BASKET',
+                    payload: {
+                        data: {
+                            product: this.categoryProducts[productIndex],
+                            count: 1
+                        }
+                    }
+                });
+                setTimeout(() => {
+                    GridViewItem.toggleIndicator(false);
+                }, 500);
+            };
+        };
+    }
+    refreshGridView() {
+        this.gvProducts.itemCount = this.categoryProducts.length;
+        this.checkIfListEmpty();
+        this.gvProducts.refreshData();
+    }
+    checkIfListEmpty() {
```

Comment:
just initialize the emptyItem once. Doing this on every refresh will be costly. Text assignment and visible dispatch can stay

Reply:
OK.